pkg/volume/csi/csi_mounter.go

```diff
@@ -272,18 +282,7 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 
 	err = c.applyFSGroup(fsType, mounterArgs.FsGroup)
 	if err != nil {
-		// attempt to rollback mount.
-		fsGrpErr := fmt.Errorf("applyFSGroup failed for vol %s: %v", c.volumeID, err)
-		if unpubErr := csi.NodeUnpublishVolume(ctx, c.volumeID, dir); unpubErr != nil {
-			klog.Error(log("NodeUnpublishVolume failed for [%s]: %v", c.volumeID, unpubErr))
-			return fsGrpErr
-		}
-
-		if unmountErr := removeMountDir(c.plugin, dir); unmountErr != nil {
-			klog.Error(log("removeMountDir failed for [%s]: %v", dir, unmountErr))
-			return fsGrpErr
-		}
-		return fsGrpErr
+		return volumetypes.NewUncertainProgressError(fmt.Sprintf("applyFSGroup failed for vol %s: %v", c.volumeID, err))
```

Comment:
maybe add comment here. At this point, mount is successful, but we want to 
1. return error so that volume cannot be used since ownership is not set correctly
2. return NewUncertainProgressError so that volume will be marked in actual state and mount will be cleaned up when it is needed.

Reply:
added comment